E2E tests use functions not available in go 1.18

# Expected Behavior

E2E tests should compile and pass when using go 1.18.x 

# Actual Behavior

The tests fail to compile because it uses one function that is not available in go 1.18. Refer:

https://github.com/tektoncd/results/blob/fdc47542cc0ce52fadaa4389a01ff0c2b75fe6e5/test/e2e/client/rest.go#L117

```sh
# github.com/tektoncd/results/test/e2e/client
client/rest.go:117:13: c.url.JoinPath undefined (type *url.URL has no field or method JoinPath)
FAIL    github.com/tektoncd/results/test/e2e [build failed]
```

Go mod file:

https://github.com/tektoncd/results/blob/fdc47542cc0ce52fadaa4389a01ff0c2b75fe6e5/go.mod#L3

Minimum go version required for this function is [1.19](https://pkg.go.dev/net/url@go1.19#URL.JoinPath) 

# Steps to Reproduce the Problem

1. Pull the latest commit 
2. Run `go version`. It should be 1.18.x
3. Run `./test/e2e-tests.sh`
4. See the error

# Additional Info
- Refer: [net/url package documentation for go1.18.10](https://pkg.go.dev/net/url@go1.18.10)
- Refer: [net/url package documentation for go1.19](https://pkg.go.dev/net/url@go1.19#URL.JoinPath) 


<!-- Any other additional information -->

--- a/docs/DEVELOPMENT.md
+++ b/docs/DEVELOPMENT.md
@@ -5,7 +5,7 @@
 The following tools are used by the project:
 
 - [git](https://git-scm.com/doc)
-- [go](https://golang.org/doc/install)
+- [go](https://golang.org/doc/install) **>=go1.19**
 - [kind](https://kind.sigs.k8s.io)
 - [ko](https://github.com/google/ko)
 - [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/)

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/tektoncd/results
 
-go 1.18
+go 1.19
 
 require (
 	github.com/aws/aws-sdk-go-v2 v1.17.8

